File: C#/datum.cs
using System;

public class datum {
    public static void Main (string[] args) {
        string line = Console.ReadLine ();
        int space = line.IndexOf (' ');
        int day = int.Parse (line.Substring (0, space));
        int month = int.Parse (line.Substring (space + 1));
        for (int x = 1; x < month; x++) {
            switch (x) {
                case 1:
                case 3:
                case 5:
                case 7:
                case 8:
                case 10:
                case 12:
                    day += 31;
                    break;
                case 2:
                    day += 28;
                    break;
                case 4:
                case 6:
                case 9:
                case 11:
                    day += 30;
                    break;
            }
        }
        day %= 7;
        Console.WriteLine (day == 0 ? "Wednesday" : day == 1 ? "Thursday" : day == 2 ? "Friday" : day == 3 ? "Saturday" : day == 4 ? "Sunday" : day == 5 ? "Monday" : "Tuesday");
    }
}

File: Java/datum.java

// https://open.kattis.com/problems/datum
import java.util.HashMap;
import java.util.Scanner;

public class datum {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        HashMap<Integer, Integer> months = new HashMap<Integer, Integer>() {
            {
                put(1, 31);
                put(2, 28);
                put(3, 31);
                put(4, 30);
                put(5, 31);
                put(6, 30);
                put(7, 31);
                put(8, 31);
                put(9, 30);
                put(10, 31);
                put(11, 30);
                put(12, 31);
            }
        };
        HashMap<Integer, String> dayOfWeek = new HashMap<Integer, String>() {
            {
                put(0, "Wednesday");
                put(1, "Thursday");
                put(2, "Friday");
                put(3, "Saturday");
                put(4, "Sunday");
                put(5, "Monday");
                put(6, "Tuesday");
            }
        };

        int day = sc.nextInt();
        int month = sc.nextInt();
        if (month != 1)
            for (int x = 1; x < month; x++)
                day += months.get(x);
        System.out.println(dayOfWeek.get(day % 7));
        sc.close();
    }
}